core/jazz_metrics/test/test.js

```diff
@@ -1,11 +1,1022 @@
+
+        sinon.assert.calledOnce(genericValidation);
+        sinon.assert.calledOnce(validateGeneralFields);
+        sinon.assert.calledOnce(getToken);
+        sinon.assert.calledOnce(getAssetsDetails);
+        sinon.assert.calledOnce(validateAssets);
+        sinon.assert.calledOnce(getMetricsDetails);
+
+        genericValidation.restore();
+        validateGeneralFields.restore();
+        getToken.restore();
+        getAssetsDetails.restore();
+        validateAssets.restore();
+        getMetricsDetails.restore();
+      });
+    });
 
-    // Add your test cases here.
-    assert(true);
-    done();
   });
+
+  describe("utils", () => {
+    it("should massage data for provided input params", () => {
+      var resObj = utils.massageData("assetData", event.body)
+      expect(resObj).to.have.all.keys('service', 'domain', 'environment', 'start_time', 'end_time', 'assets', 'interval', 'statistics');
+      expect(resObj.assets).to.eq("assetData");
+    });
+
+    it("should re-arrange asset data for provided input/result data", () => {
+      var results = [{
+        "Label": "Duration",
+        "Datapoints": [{
+          "Timestamp": "2018-06-28T10:07:00.000Z",
+          "Sum": 29.78,
+          "Unit": "Milliseconds"
+        }]
+      }];
+      var assetItem = {
+        type: 's3',
+        statistics: 'sum',
+        asset_name: 'assetName'
+      }
+      var resObj = utils.assetData(results, assetItem);
+      expect(resObj).to.have.all.keys('type', 'asset_name', 'statistics', 'metrics');
+    });
+
+    describe('getNameSpaceAndMetricDimensons', () => {
+      it("should provide all metrics params  for provided namespace", () => {
+        var validNameSpace = ['apigateway', 'cloudfront', 'lambda', 's3'];
+        validNameSpace.forEach(namespace => {
+          var resObj = utils.getNameSpaceAndMetricDimensons(namespace);
+          expect(resObj).to.have.all.keys('isError', 'paramMetrics', 'awsNameSpace');
+          expect(resObj.isError).to.be.false;
+          resObj.paramMetrics.forEach(each => {
+            if (namespace === 's3') {
+              expect(each).to.have.all.keys('MetricName', 'Unit', 'Dimensions', 'Statistics');
+            } else {
+              expect(each).to.have.all.keys('MetricName', 'Unit', 'Dimensions');
+            }
+          })
+        });
+      });
+
+      it("should indicate error while accessing metric params for invalid namespace", () => {
+        var invalidNameSpace = ['swagger_url', 'endpoint_url'];
+        invalidNameSpace.forEach(namespace => {
+          var resObj = utils.getNameSpaceAndMetricDimensons(namespace);
+          expect(resObj).to.have.all.keys('isError', 'awsNameSpace');
+          expect(resObj.isError).to.be.true;
+          expect(resObj.awsNameSpace).to.eq("Invalid");
+        })
+      })
+    });
+
+    it("should successfully get asset object for provided asset details", () => {
```

Comment:
How about validating for an asset type that is not supported yet? Send asset_type as `foo` and the API should return appropriate response (not supported)?

Reply:
Included validation for asset type swagger_url and endpoint_url (Doesn't support for metrics).